pkg/client/openshift/buildconfig.go

```diff
@@ -101,6 +111,37 @@ func (b *buildConfig) TriggerBuild(bc *buildv1.BuildConfig, triggedBy string) (b
 	return true, nil
 }
 
+// TriggerBuildFromFile wil be called by kogito-cli when a build from file is performed.
+// When called a new build will be triggered with the request kogito resource or a tgz file.
+func (b *buildConfig) TriggerBuildFromFile(namespace string, bodyPost io.Reader, options *buildv1.BinaryBuildRequestOptions) (*buildv1.Build, error) {
+
+	result := &buildv1.Build{}
+	buildName := fmt.Sprintf("%s%s", options.Name, "-builder")
+
+	// before upload the file, make sure that the build exist
+	err := b.waitForBuildConfig(checkBcRetries, checkBcRetriesInterval, func() (err error) {
+		if targetBuildConfig, err := b.client.BuildCli.BuildConfigs(namespace).Get(buildName, metav1.GetOptions{}); err != nil && errors.IsNotFound(err) {
+			log.Debugf("BuildConfig %s not found in the %s namespace", targetBuildConfig.Name, targetBuildConfig.Namespace)
+			return err
+		}
+		return
+	})
+	if err != nil {
+		log.Error(err)
```

Comment:
What is the reason to log an error here and continue?
Should build be triggered even when `waitForBuildConfig` ends in error?

Reply:
+1